.github/workflows/specialized-test-runner.yml

```diff
@@ -100,7 +108,7 @@ jobs:
       fail-fast: false
       matrix:
         tests: ${{ fromJson(needs.generate_tests_matrix.outputs.runsheet) }}
-    if: ${{ github.repository_owner == 'dotnet' }}
+    if: ${{ always() && github.repository_owner == 'dotnet' && needs.generate_tests_matrix.result == 'success' && (needs.build_packages.result == 'success' || needs.build_packages.result == 'skipped') }}
```

Comment:
why `always()` here?

Reply:
that's not correct. Good catch, I'll fix.